client/src/main/com/sinch/sdk/domains/conversation/adapters/ContactService.java

```diff
@@ -0,0 +1,159 @@
+import com.sinch.sdk.models.Configuration;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+
+public class ContactService implements com.sinch.sdk.domains.conversation.ContactService {
+
+  private final Configuration configuration;
+  private final ContactApi api;
+
+  private final Collection<ChannelType> supportedChannelForGetProfile =
+      Arrays.asList(
+          ChannelType.INSTAGRAM, ChannelType.MESSENGER, ChannelType.VIBER, ChannelType.LINE);
+
+  public ContactService(
+      Configuration configuration, HttpClient httpClient, Map<String, AuthManager> authManagers) {
+    this.configuration = configuration;
+    this.api =
+        new ContactApi(
+            httpClient, configuration.getConversationServer(), authManagers, new HttpMapper());
+  }
+
+  protected ContactApi getApi() {
+    return this.api;
+  }
+
+  public ContactListResponse list(ContactListRequestParameters iclient) throws ApiException {
+    ContactListRequestParametersImpl client = (ContactListRequestParametersImpl) iclient;
+
+    if (null == client) {
+      client = ContactListRequestParametersImpl.builder(null).build();
+    }
+
+    String externalId = client.getExternalId().orElse(null);
+    ConversationChannelDto channel = ChannelTypeDtoConverter.convert(client.getChannel());
+    String identity = client.getIdentity().orElse(null);
+
+    Integer pageSize = client.getPageSize().orElse(null);
+    String pageToken = client.getPageToken().orElse(null);
+
+    ListContactsResponseDto response =
+        getApi()
+            .contactListContacts(
+                configuration.getProjectId(), pageSize, pageToken, externalId, channel, identity);
+
+    Pair<Collection<Contact>, TokenPageNavigator> content = ContactDtoConverter.convert(response);
+
+    return new ContactListResponse(
+        this, new Page<>(client, content.getLeft(), handleEmptyPageToken(content.getRight())));
+  }
+
+  public Contact get(String contactId) {
+    return ContactDtoConverter.convert(
+        getApi().contactGetContact(configuration.getProjectId(), contactId));
+  }
+
+  public Contact create(ContactCreateRequestParameters iclient) {
+    ContactCreateRequestParametersImpl client = (ContactCreateRequestParametersImpl) iclient;
+
+    return ContactDtoConverter.convert(
+        getApi()
+            .contactCreateContact(
+                configuration.getProjectId(), ContactDtoConverter.convertForCreate(client)));
+  }
+
+  public void delete(String contactId) {
+
+    getApi().contactDeleteContact(configuration.getProjectId(), contactId);
+  }
+
+  public Contact update(Contact iclient) {
+    ContactImpl client = (ContactImpl) iclient;
+
+    return ContactDtoConverter.convert(
+        getApi()
+            .contactUpdateContact(
+                configuration.getProjectId(),
+                client.getId(),
+                ContactDtoConverter.convert(client),
+                null));
```

Comment:
About updateMask: still no need to be implemented. Without mask, backend will handle only fields present from payload so, mask is not required to update any field.
About function prototype: thisd is the call to "underlying API interface " generated by OAS generator tool; so this file is fully in it "adaptor's role": it hide the mask usage:
-  SDK's user is calling `service.update()` without any mask reference just the `Contact` instance with fields to be updated
- adaptor is calling underlying API with a null mask value

Reply:
Ok, so if I understand well, we can state that the Java SDK is not supporting one of the API feature which is to send an `update_mask` along with the update request body, right?

Reply:
Exact: due to Java compilation and struct this is not required:
- objects are immutable: so you cannot just get an object and edit one a dedicated field to send back with the mas to server
- you have to create a new object with the field you want to edit and the value you want to change: so at this point it replace the requirement to set a mask in addition to set the only required field
- because of SDK is aware to detect which fields was really set onto object instance (thank to OptionalValue dedicated class); it just sent to server fields to be modified